File: ts/examples/simple-example-ts/src/functions/fibonacciRuntimeFunction.ts
import {
    DisplayMessage,
    FunctionContext, Parameter,
    Identifier,
    Name,
    OmitRuntimeFunction,
    Signature,
} from "@code0-tech/hercules";

@Identifier("fibonacci_runtime")
@Signature("(test: number): number")
@Name({code: "en-US", content: "Fibonacci (Runtime)"})
@DisplayMessage({code: "en-US", content: "Computes the n-th Fibonacci number"})
@OmitRuntimeFunction()
@Parameter({runtimeName: "test", name: [{code: "en-US", content: "N"}]})
export class FibonacciRuntimeFunction {
    run(context: FunctionContext, test: number): number {
        console.log(`[fibonacci] project=${context.projectId} execution=${context.executionId}`);
        return this.fib(test);
    }

    private fib(n: number): number {
        if (n <= 1) return n;
        return this.fib(n - 1) + this.fib(n - 2);
    }
}

File: py/examples/simple-example-py/functions/fibonacci_runtime_function.py
from hercules import (
    DisplayMessage,
    FunctionContext,
    Identifier,
    Name,
    OmitRuntimeFunction,
    Parameter,
    RuntimeFunctionRunnable,
    Signature,
)


@Identifier("fibonacci_runtime")
@Signature("(test: number): number")
@Name({"code": "en-US", "content": "Fibonacci (Runtime)"})
@DisplayMessage({"code": "en-US", "content": "Computes the n-th Fibonacci number"})
@OmitRuntimeFunction()
@Parameter({"runtime_name": "test", "name": [{"code": "en-US", "content": "N"}]})
class FibonacciRuntimeFunction(RuntimeFunctionRunnable):
    def run(self, context: FunctionContext, test):
        print(f"[fibonacci] project={context.project_id} execution={context.execution_id}")
        return self._fib(test)

    def _fib(self, n):
        if n <= 1:
            return n
        return self._fib(n - 1) + self._fib(n - 2)
